video/src/ai/eto/rikai/sql/spark/datasources/VideoOptions.scala

```diff
@@ -1,21 +1,65 @@
 package ai.eto.rikai.sql.spark.datasources
 
 import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.bytedeco.ffmpeg.global.swscale
+import org.bytedeco.javacv.FrameGrabber
 
 class VideoOptions(@transient val parameters: CaseInsensitiveMap[String])
     extends Serializable {
   def this(parameters: Map[String, String]) = {
     this(CaseInsensitiveMap(parameters))
   }
 
+  // Number of frames per second
   val fps = parameters.get("fps").map(_.toInt).getOrElse(1)
+  // Frame step size and the frame_id to start
+  val frameStepSize = parameters.get("frameStepSize").map(_.toInt).getOrElse(0)
```

Comment:
How about using
```
  val frameStepSize : Option[Int] = parameters.get("frameStepSize").map(_.toInt)
  val frameStartId : Option[Int] = parameters.get("frameStartId").map(_.toInt)
```
`0` might be a valid value for some parameters, so we need to distinguish between "user have not filled this parameter" and "user fill this parameter with 0", is it a valid value for `frameStartId`?

Reply:
in this case, 0 is a valid parameter

Reply:
parameters.get("xxx") is already an Option. 0 is a default value here.

Reply:
Let me add more inline comments